src/ucp/rndv/proto_rndv.c

```diff
@@ -708,6 +708,32 @@ void ucp_proto_rndv_bulk_query(const ucp_proto_query_params_t *params,
     ucp_proto_multi_query_config(&multi_query_params, attr);
 }
 
+/* Sys device is crucial for rkey_unpack in rndv pipeline protocols that use
+ * CUDA staging buffers. If sys_dev of the user memory is unknown (e.g. host
+ * or CUDA-managed memory), use the default sys_dev since it will be used for
+ * allocating staging buffers anyway. */
```

Comment:
Could we say
If we don't know the system device of the user buffer, unpack the remote key on the system device that pipeline protocols would use for staging.
and rename function to
ucp_proto_rndv_ppln_frag_sys_dev